sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVParameters.scala

```diff
@@ -73,6 +82,12 @@ private[sql] case class CSVParameters(@transient parameters: Map[String, String]
 
   val nullValue = parameters.getOrElse("nullValue", "")
 
+  val compressionCodec: String = {
```

Comment:
maybe we should do some data validation here, i.e. throwing exceptions if class not found.